### PR TITLE
Rover: send WHEEL_DISTANCE messages

### PR DESCRIPTION
**Pull request by rmackay9**
This is a replacement for [this PR](https://github.com/ArduPilot/ardupilot/pull/10219) from @pavloblindnology that sends the [WHEEL_DISTANCE](https://mavlink.io/en/messages/common.html#WHEEL_DISTANCE) mavlink message

- this message is sent as part of the EXTRA3 stream but the MESSAGE_INTERVAL feature should also work
- the timestamp sent in the message is the time the data was pushed to the EKF.  It's possible that this should instead be the most recent ping from a wheel encoder (i.e. wheel_encoder_last_update_ms) but it's probably not too important
- changes from the previous PR are:
  - std::copy is replaced with a loop to copy data into the distances array before sending.  std::copy is not normally used in AP (memcpy is) but in any case, the wheel-distance message wants doubles but we store the wheel encoder distances as floats so I don't think a memcpy really works.  We only support 2 wheel encoders so the performance impact is very small in any case.
  - formatting fixes

### CODE REVIEW

**Comment by rmackay9** [comment]
@pavloblindnology, does this PR look ok to you?

File: APMrover2/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -287,6 +287,18 @@ void Rover::send_wheel_encoder(mavlink_channel_t chan)
     }
 }
 
+void Rover::send_wheel_encoder_distance(mavlink_channel_t chan)
+{
+    // send wheel encoder data using wheel_distance message
+    if (g2.wheel_encoder.num_sensors() > 0) {
+        double distances[16] {};
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply APMrover2/GCS_Mavlink.cpp]: @pavloblindnology, thanks for checking this PR.  We can put that bit of code in there but in practice the size of the array in mavlink can't be changed.  Once a mavlink message is defined, if it's ever changed then any older software using the old .xml file can't read the new message.. so they're never changed.

[the remark this answers is not in the document]
Comment (pavloblindnology) [reply APMrover2/GCS_Mavlink.cpp]: @rmackay9 Got it, thanks.

**Comment by pavloblindnology** [comment]
LGTM.

File: APMrover2/GCS_Mavlink.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (khancyr) [reply APMrover2/GCS_Mavlink.cpp]: can't we have a constant for this 16 ? like MAVLINK_MSG_WHEEL_DISTANCE_FIELD_DISTANCE_LEN ?

[the remark this answers is not in the document]
Comment (rmackay9) [reply APMrover2/GCS_Mavlink.cpp]: @khancyr, aha, I didn't know about that definition, added, thanks!

**Comment by rmackay9** [comment]
Merged, thanks for the reviews!

**Comment by pavloblindnology** [comment]
When is this going to be included in some stable release?

**Comment by rmackay9** [comment]
It missed the Rover-3.5 release so by default it wouldn't normally be released until Rover-3.6 but we could make an exception and try and include this in Rover-3.5.1 which may start beta testing in the next week or so.  I've added it to the list of things to add to Rover-3.5.1

**Comment by pavloblindnology** [comment]
Thanks.

**Comment by rmackay9** [comment]
I'm finally adding this to Rover-3.5.1-rc1 which I hope will start beta testing this week.

**Pull request merged**